aws-sdk-resources/features/s3/download_file.feature

```diff
@@ -0,0 +1,57 @@
+# language: en
+@s3 @resources @downloadfile
+Feature: Managed file download
+
+ Background:
+    Given I create a bucket
+
+  @get
+  Scenario: Download a small object
+    Given I have a 1M file
+    And I upload the file using put_object
+    When I download the file with mode "auto"
+    And 1 get_object request should have been made
+    And the downloaded file should match the uploaded file
+
+  # Any object uploaded in parts, is downloaded in parts
+  @get-part
+  Scenario: Download a large object using part numbers
+    Given I have a 16M file
+    And I upload the file
+    When I download the file with mode "auto"
+    Then 4 upload_part requests should have been made
+    And 4 get_object requests should have been made
+    And the downloaded file should match the uploaded file
+
+  # "larger" objects (exceeding 15MB) are downloaded in chunks,
+  # using RANGE requests, even if the it was not uploaded in parts
+  @get-range
+  Scenario: Download a large object using ranges
+    Given I have a 16M file
+    And I upload the file using put_object
+    When I download the file with mode "auto"
+    Then 4 get_object requests should have been made
+    And the downloaded file should match the uploaded file
+
+  # Users can specify mode: 'single-request' to force the object
+  # to download in 1 request, this avoids the HEAD object request
+  # penalty, especially useful for small objects.`
+  @mode
+  Scenario: Download a large object using part numbers
+    Given I have a 1M file
+    And I upload the file
+    When I download the file with mode "single_request"
+    Then 0 head_object request should have been made
+    And 1 get_object request should have been made
+    And the downloaded file should match the uploaded file
+
+  # It is possible for the user to control the chunk size of each
+  # part downloaded, by specifing a mode "get-range" and also
+  # an option to define how large each chunk is.
+  @mode
+  Scenario: Download a large object using part numbers
+    Given I have a 16M file
```

Comment:
Now that we are creating large files, I'm a bit concerned that we might be leaking tempfiles with how our tests are written. We should add explicit cleanup steps and a shared tag for these tests like `@large-file` or similar.